pkg/containerimagebuilderpusher/kaniko.go

```diff
@@ -0,0 +1,219 @@
+	defer os.Remove(assetPath) // nolint: errcheck
+
+	// Generate kaniko job spec
+	kanikoJobSpec := k.getKanikoJobSpec(buildOptions, bundleFilename)
+
+	k.logger.DebugWith("About to publish kaniko job", "jobSpec", kanikoJobSpec)
+	kanikoJob, err := k.kubeClientSet.BatchV1().Jobs(namespace).Create(kanikoJobSpec)
+	if err != nil {
+		return errors.Wrap(err, "Failed to publish kaniko job")
+	}
+
+	k.logger.Debug("Waiting for kaniko to finish")
+	timeout := time.Now().Add(10 * time.Minute)
+	for time.Now().Before(timeout) {
+		runningJob, err := k.kubeClientSet.BatchV1().Jobs(namespace).
+			Get(kanikoJob.Name, meta_v1.GetOptions{IncludeUninitialized: true})
+
+		if err != nil {
+			return errors.Wrap(err, "Failed to poll kaniko job status")
+		}
+
+		if runningJob.Status.Succeeded > 0 {
+			k.logger.Debug("Kaniko job was completed successfully")
+
+			// Cleanup
+			propagationPolicy := meta_v1.DeletePropagationBackground
+			err = k.kubeClientSet.BatchV1().Jobs(namespace).Delete(kanikoJob.Name, &meta_v1.DeleteOptions{
+				PropagationPolicy: &propagationPolicy,
+			})
+			if err != nil {
+				k.logger.Error("Failed to delete Kaniko job after successful completion")
+			}
+			return nil
+		}
+		if runningJob.Status.Failed > 0 {
+			k.logger.DebugWith("Kaniko job has failed", "status", runningJob.Status)
+			return errors.New("Kaniko job has failed")
+		}
+	}
+	return errors.New("Kaniko job has timed out")
+}
+
+func (k *Kaniko) createDockerBuildBundle(image string, contextDir string, tempDir string) (string, string, error) {
+	var err error
+
+	// Create temp directory to store compressed docker build bundle
+	buildDockerBundleDir := path.Join(tempDir, "tar")
+	err = os.Mkdir(buildDockerBundleDir, 0744)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "Failed to create tar dir: %s", buildDockerBundleDir)
+	}
+
+	k.logger.DebugWith("Created tar dir", "dir", buildDockerBundleDir)
+
+	tarFilename := fmt.Sprintf("%s.tar.gz", strings.Replace(image, "/", "_", -1))
+	tarFilename = strings.Replace(tarFilename, ":", "_", -1)
+	tarFilePath := path.Join(buildDockerBundleDir, tarFilename)
+
+	k.logger.DebugWith("Compressing build bundle", "tarFilePath", tarFilePath)
+
+	// Just in case file with the same filename already exist - delete it
+	_ = os.Remove(tarFilePath)
+
+	_, err = exec.Command("tar", "-zcvf", tarFilePath, contextDir).Output()
+	if err != nil {
+		return "", "", errors.Wrapf(err, "Failed to compress build bundle")
+	}
+	k.logger.Debug("Build bundle was successfully compressed")
+
+	// Create symlink to bundle tar file in nginx serving directory
+	assetPath := path.Join("/etc/nginx/static/assets", tarFilename)
+	err = os.Link(tarFilePath, assetPath)
+	if err != nil {
+		return "", "", errors.Wrapf(err, "Failed to create symlink to build bundle")
+	}
+
+	return tarFilename, assetPath, nil
+}
+
+func (k *Kaniko) getKanikoJobSpec(buildOptions *BuildOptions, bundleFilename string) *batch_v1.Job {
+
+	completions := int32(1)
```

Comment:
why not simply pass as number?

Reply:
the required format is pointer to an int
so the int must be created first